pallets/liquid-staking/src/lib.rs

```diff
@@ -1732,6 +1753,37 @@ pub mod pallet {
             Ok(())
         }
 
+        // liquid_amount_to_fee=TotalLiquidCurrency * (commission_rate*total_rewards/(TotalStakeCurrency+(1-commission_rate)*total_rewards))
+        fn do_charge_commission(rewards: BalanceOf<T>) -> DispatchResult {
+            let liquid_currency = Self::liquid_currency()?;
+            let issuance = T::Assets::total_issuance(liquid_currency);
+            let commission_rate = Self::commission_rate();
+            if issuance.is_zero() || commission_rate.is_zero() {
+                return Ok(());
+            }
+
+            let matching_ledger = Self::matching_pool();
+            let total_active_bonded = Self::get_total_active_bonded();
+            let total_stake_amount = total_active_bonded
+                .checked_add(matching_ledger.total_stake_amount.total)
+                .and_then(|r| r.checked_sub(matching_ledger.total_unstake_amount.total))
+                .ok_or(ArithmeticError::Overflow)?;
+
+            let numerator = commission_rate
+                .saturating_mul_int(rewards)
+                .saturating_mul(issuance);
+            let denominator = total_stake_amount
+                .saturating_add(rewards)
+                .saturating_sub(commission_rate.saturating_mul_int(rewards));
+            let liquid_amount_to_fee = numerator
+                .checked_div(denominator)
+                .unwrap_or_else(Zero::zero());
+
+            T::Assets::mint_into(liquid_currency, &Self::account_id(), liquid_amount_to_fee)?;
```

Comment:
this hasn't been recorded into `TotalReserves`

Reply:
`TotalReserves` is for staking currency, however here is liquid currency